src/lib/features/like-episode/model/like.ts

```diff
@@ -0,0 +1,51 @@
+import type { Episode } from '$lib/shared/api';
+import { persistentWritable, localStorageAdapter } from 'svelte-persistent-writable';
+
+/**
+ * The store containing set of episodes liked by the current user.
+ */
+export const likesStore = persistentWritable(new Map<Episode['id'], Episode>(), {
+  storage: localStorageAdapter('likes', {
+    serialize(likes: Map<Episode['id'], Episode>) {
+      const episodes: Episode[] = Array.from(likes.values());
+      return JSON.stringify(episodes);
+    },
+    deserialize(serialized: string) {
+      const episodes: Episode[] = JSON.parse(serialized);
+      return new Map(episodes.map((episode) => [episode.id, episode]));
+    },
+  }),
+});
+
+/**
+ * Register that the user liked the given episode.
+ * @param episode The id of the episode to be liked.
+ */
+export function addLike(episode: Episode) {
+  likesStore.update((likes) => likes.set(episode.id, episode));
+}
+
+/**
+ * Remove the like for the given episode.
+ * @param episode The id of the episode whose like is to be removed.
+ */
+export function removeLike(episode: Episode) {
+  likesStore.update((likes) => (likes.delete(episode.id), likes));
```

Comment:
[issue](https://conventionalcomments.org/#labels) (non-blocking): however elegant, this snippet is exotic, not very familiar.

[suggestion](https://conventionalcomments.org/#labels) (non-blocking): could we gain some code clarity by being explicit and writing out a return here instead?